lib/controllers.js

```diff
@@ -517,6 +518,88 @@ module.exports = function(rootDirectory, logger, indexDocument, errorDocument) {
             deleteObjects(req, res);
           })
         );
+      } else if (
+        req.headers["content-type"].indexOf("multipart/form-data") !== -1
+      ) {
+        const fields = {
+          bucket: req.params.bucket
+        };
+        const form = new multiparty.Form();
+        let chain = Promise.resolve();
+        let fileWasFound = false;
+
+        form.on("error", err => {
+          res.status(400).send(err);
+        });
+
+        form.on("part", part => {
+          if (!part.filename) {
+            chain = chain.then(() => {
+              return new Promise(resolve => {
+                let string = "";
+                part
+                  .on("data", buf => (string += buf.toString()))
+                  .on("end", () => {
+                    fields[part.name] = string;
+                    resolve();
+                  });
+              });
+            });
+          } else {
+            // 'file' must be the last field in the post (this is how aws requires it)
```

Comment:
Can you elaborate on this comment? I don't see any reference to a `file` field in the code